js/map.js

```diff
@@ -38,22 +71,43 @@
 
   window.util.setDisabled(mapFiltersFormSelect);
 
-  var onTypeChange = function () {
-    window.card.deleteElements();
-    var currentType = mapFiltersTypeFilter.value;
+  var filtrationItem = function (filter, item, key) {
+    return filter.value === 'any' ? true : filter.value === item[key].toString();
+  };
 
+  var filtrationByType = function (item) {
+    return filtrationItem(typeFilter, item.offer, 'type');
+  };
 
-    var pinsAfterFiltering = pinsData.filter(function (it) {
-      if (currentType === 'any') {
-        return true;
-      }
-      return it.offer.type === currentType;
+  var filtrationByPrice = function (item) {
+    var currentPrice = PriceRange[priceFilter.value];
+    return currentPrice ? item.offer.price >= currentPrice.min && item.offer.price <= currentPrice.max : true;
+  };
+
+  var filtrationByRooms = function (item) {
+    return filtrationItem(roomFilter, item.offer, 'rooms');
+  };
+
+  var filtrationByGuests = function (item) {
+    return filtrationItem(guestFilter, item.offer, 'guests');
+  };
+
+  var filtrationByFeatures = function (item) {
+    var checkedFeaturesItems = featuresFilter.querySelectorAll('input:checked');
+    return Array.from(checkedFeaturesItems).every(function (element) {
+      return item.offer.features.includes(element.value);
     });
+  };
+
+  var onFilterChange = debounce(function () {
+    window.card.deleteElements();
+    filteredData = pinsData.slice(0);
+    var pinsAfterFiltering = filteredData.filter(filtrationByType).filter(filtrationByPrice).filter(filtrationByRooms).filter(filtrationByGuests).filter(filtrationByFeatures);
```

Comment:
Лучше переноси по строчкам, обычно так делают чтоб не было длинной строки
```suggestion
    var pinsAfterFiltering = filteredData.filter(filtrationByType)
        .filter(filtrationByPrice)
        .filter(filtrationByRooms)
        .filter(filtrationByGuests)
        .filter(filtrationByFeatures);
```